skxray/core.py

```diff
@@ -604,6 +605,51 @@ def bin_1D(x, y, nx=None, min_x=None, max_x=None):
     return bins, val, count
 
 
+def statistics_1D(x, y, stat='mean', nx=None, min_x=None, max_x=None):
+    """
+    Bin the values in y based on their x-coordinates
+
+    Parameters
+    ----------
+    x : array
+        position
+    y : array
+        intensity
+    stat: str or func, optional
+        statistic to be used on the binned values
+        see scipy.stats.binned_statistic
+    nx : integer, optional
+        number of bins to use
+    min_x : float, optional
+        Left edge of first bin
+    max_x : float, optional
+        Right edge of last bin
+
+    Returns
+    -------
+    edges : array
+        edges of bins, length nx + 1
+
+    val : array
+        statistics of values in each bin, length nx
+    """
+
+    # handle default values
+    if min_x is None:
+        min_x = np.min(x)
+    if max_x is None:
+        max_x = np.max(x)
+    if nx is None:
+        nx = _defaults["bins"]
+
+    # use a weighted histogram to get the bin sum
+    bins = np.linspace(start=min_x, stop=max_x, num=nx+1, endpoint=True)
```

Comment:
We have a helper function in core to take any permutation of how to specify bin edges.  Might be worth either use that here taking in the bins array as input.